eodatasets3/scripts/tostac.py

```diff
@@ -7,70 +7,182 @@
 )
-def run(odc_metadata_files: Iterable[Path]):
+def run(
+    odc_metadata_files: Iterable[Path], stac_template, stac_base_url, product_base_url
+):
     for input_metadata in odc_metadata_files:
         dataset = serialise.from_path(input_metadata)
 
-        project = partial(
-            pyproj.transform,
-            pyproj.Proj(init=dataset.crs),
-            pyproj.Proj(init="epsg:4326"),
-        )
-        wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
-        item_doc = dict(
-            id=dataset.id,
-            type="Feature",
-            bbox=wgs84_geometry.bounds,
-            geometry=wgs84_geometry.__geo_interface__,
-            properties={**dataset.properties, "odc:product": dataset.product.name},
-            assets={
-                # TODO: Currently assuming no name collisions.
-                **{name: {"href": m.path} for name, m in dataset.measurements.items()},
-                **{name: {"href": m.path} for name, m in dataset.accessories.items()},
-            },
-            links=[
-                # {
-                #     "rel": "self",
-                #     "href": '?',
-                # },
-                {"rel": "odc_product", "href": dataset.product.href},
-                {
-                    "rel": "alternative",
-                    "type": "text/html",
-                    "href": f"https://explorer.dea.ga.gov.au/dataset/{dataset.id}",
-                },
-            ],
-        )
-
         name = input_metadata.stem.replace(".odc-metadata", "")
         output_path = input_metadata.with_name(f"{name}.stac-item.json")
 
+        # Load STAC template data
+        if stac_template:
+            with open(stac_template, "r") as f:
+                stac_data = json.load(f)
+        else:
+            stac_data = {}
+
+        # Create STAC dict
+        item_doc = create_stac(
+            dataset,
+            input_metadata,
+            output_path,
+            stac_data,
+            stac_base_url,
+            product_base_url,
+        )
+
         with output_path.open("w") as f:
             json.dump(item_doc, f, indent=4, default=json_fallback)
 
-        echo(output_path)
+
+def create_stac(
+    dataset, input_metadata, output_path, stac_data, stac_base_url, product_base_url
+):
+    """
+    Create STAC document
+
+    :param dataset: Dict of the metadata content
+    :param input_metadata: Path of the Input metadata file
+    :param output_path: Path of the STAC output file
+    :param stac_data: Dict of the static STAC content
+    :param stac_base_url: Base URL for STAC file
+    :param product_base_url: Base URL for ODC product
+    :return: Dict of the STAC content
+    """
+
+    project = partial(
+        pyproj.transform, pyproj.Proj(init=dataset.crs), pyproj.Proj(init="epsg:4326"),
+    )
+    wgs84_geometry: BaseGeometry = transform(project, dataset.geometry)
+    item_doc = dict(
+        stac_version=stac_data.get("stac_version", "1.0.0-beta.1"),
```

Comment:
I'm not sure about us taking stac version from both sources. 

This script adds properties of a specific version of stac, and if someone else's template uses a different version of stac there will be an inconsistent result. I think I'd prefer just to throw an error if their template doesn't use the same version as this script. (... sometime in the future, if needed, we could always add a parameter for choosing stac version)

Reply:
Added STAC version check